blivet/devicetree.py

```diff
@@ -150,12 +150,18 @@ def _add_device(self, newdev, new=True):
             :param newdev: the device to add
             :type newdev: a subclass of :class:`~.devices.StorageDevice`
 
-            Raise ValueError if the device's identifier is already
+            Raise DeviceTreeError if the device's identifier is already
             in the list.
         """
         if newdev.uuid and newdev.uuid in [d.uuid for d in self._devices] and \
            not isinstance(newdev, NoDevice):
-            raise ValueError("device is already in tree")
+            # Just found a device with already existing UUID. Is it the same device?
+            dev = self.get_device_by_uuid(newdev.uuid, incomplete=True, hidden=True)
+            if dev.name == newdev.name:
+                raise DeviceTreeError("Trying to add already existing device.")
```

Comment:
We could keep the `ValueError` here and change `DuplicateUUIDError` to inherit from both `UnusableConfigurationError` and `ValueError` to avoid the exception type change. This should enough to avoid an API break. @dwlehman

Reply:
Sounds like a plan to me.

Reply:
Fixed.